src/statisticalmodel.jl

```diff
@@ -170,11 +170,14 @@ Indicate whether the model has been fitted.
 function isfitted end
 
 """
+    fit(model::StatisticalModel)
```

Comment:
This method signature isn't quite right; packages typically define `fit(::Type{M}, args...; kwargs...)` where `M` is a `StatisticalModel` subtype defined in the package. However, I believe we avoided specifying an expected method signature in the docstring in order to leave it open-ended for packages that deviate for whatever reason. (Perhaps @nalimilan can correct me if I'm mistaken there.)

Reply:
I'm not sure there's a real intention behind the absence of the signature in the docstring. I would think the main reason is that there's no signature imposed, other than the first argument being a type as @ararslan said. Maybe documenting that would be useful at least? But we should check on JuliaHub that packages follow that minimal convention.